From c62f1400a4f7a4686a1030c99b1cc1f40720c9dc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Wed, 28 Dec 2022 12:23:16 +0100
Subject: [PATCH] meson: Install various missing files

While switching our Alpine Linux radare2 package from the GNU autotools
build system to the meson build system I noticed that the meson build
system doesn't install several files which were previously installed by
the GNU autotools buildsystem. This patch adds the missing files to the
corresponding meson.build files.
---
 libr/anal/d/meson.build | 1 +
 libr/asm/d/meson.build  | 2 ++
 libr/bin/d/meson.build  | 4 ++++
 libr/meson.build        | 4 +++-
 meson.build             | 1 +
 shlr/sdb/meson.build    | 3 ++-
 6 files changed, 13 insertions(+), 2 deletions(-)

diff --git a/libr/anal/d/meson.build b/libr/anal/d/meson.build
index 6198f6021c..62136d36ca 100644
--- a/libr/anal/d/meson.build
+++ b/libr/anal/d/meson.build
@@ -10,6 +10,7 @@ sdb_files = [
   'cc-ppc-64',
   'cc-riscv-64',
   'cc-sparc-32',
+  'cc-hexagon-32',
   'cc-v850-32',
   'cc-s390-64',
   'cc-x86-16',
diff --git a/libr/asm/d/meson.build b/libr/asm/d/meson.build
index e9db18749d..b5f58a31bf 100644
--- a/libr/asm/d/meson.build
+++ b/libr/asm/d/meson.build
@@ -6,6 +6,7 @@ sdb_files = [
   'avr',
   'bf',
   'bpf',
+  'chip8',
   'dalvik',
   'i4004',
   'i8080',
@@ -18,6 +19,7 @@ sdb_files = [
   'pic18c',
   'ppc',
   'propeller',
+  'pickle',
   'riscv',
   'sh',
   'sparc',
diff --git a/libr/bin/d/meson.build b/libr/bin/d/meson.build
index e1b2d183b2..177c2174eb 100644
--- a/libr/bin/d/meson.build
+++ b/libr/bin/d/meson.build
@@ -14,6 +14,8 @@ sdb_files = [
   'iertutil',
   'kernel32',
   'mfc120',
+  'mfc100u',
+  'mfc140u',
   'mfc30',
   'mfc40',
   'mfc42',
@@ -176,6 +178,8 @@ foreach file : sdb_files
 endforeach
 
 format_files = [
+  'dex',
+  'macho',
   'elf32',
   'elf64',
   'elf_enums',
diff --git a/libr/meson.build b/libr/meson.build
index 1d1fbafe2f..f400dc7c16 100644
--- a/libr/meson.build
+++ b/libr/meson.build
@@ -439,6 +439,7 @@ include_files = [
   'include/r_core.h',
   'include/r_crypto.h',
   'include/r_debug.h',
+  'include/r_drx.h',
   'include/r_egg.h',
   'include/r_endian.h',
   'include/r_flag.h',
@@ -534,7 +535,8 @@ r_util_files = [
   'include/r_util/r_utf8.h',
   'include/r_util/r_w32.h',
   'include/r_util/r_w32dw.h',
-  'include/r_util/r_x509.h'
+  'include/r_util/r_x509.h',
+  'include/r_util/r_xml.h'
 ]
 install_headers(r_util_files, install_dir: join_paths(r2_incdir, 'r_util'))
 
diff --git a/meson.build b/meson.build
index a63aa1a259..f8ea7939b1 100644
--- a/meson.build
+++ b/meson.build
@@ -676,6 +676,7 @@ if cli_enabled
     'man/ragg2.1',
     'man/rahash2.1',
     'man/rarun2.1',
+    'man/rasign2.1',
     'man/rasm2.1',
     'man/rax2.1',
     'man/ravc2.1',
diff --git a/shlr/sdb/meson.build b/shlr/sdb/meson.build
index 3cb3cf7efd..9b5eb5becb 100644
--- a/shlr/sdb/meson.build
+++ b/shlr/sdb/meson.build
@@ -117,7 +117,8 @@ if not meson.is_subproject()
     'include/sdb/sdb.h',
     'include/sdb/ht.h',
     'include/sdb/set.h',
-    'include/sdb/types.h'
+    'include/sdb/types.h',
+    'include/sdb/rangstr.h'
   ]
   install_headers(include_files, subdir: 'sdb')
 endif
